thavalon-webapp/src/components/Login.tsx

```diff
@@ -26,18 +40,29 @@ function Login(props: LoginProps) {
         event.preventDefault();
     }
 
-    function OnSubmit(data: any, event: any) {
-        console.log("SUBMIT");
-        console.log(data);
+    async function OnSubmit(data: LoginData, event: any) {
+        setDisabled(true);
 
-        // try logging in, if it works update page to reflect that
-        let loggedIn = log_in();
-        useEffect(() => props.setLoggedIn(log_in()));
+        const accountManager: AccountManager = AccountManager.getInstance();

```

Comment:
tioli: Does getInstance explicitly type its return? If so, we don't need the `: AccountManager` type here. I find it easier to read with less explicit types, but up to you.